src/Database/Query.php

```diff
@@ -1259,9 +1259,13 @@ public function orHaving($conditions, $types = [])
      * @param int|null $limit The number of rows you want in the page. If null
      *  the current limit clause will be used.
      * @return $this
+     * @throws \InvalidArgumentException If page number < 1.
      */
     public function page($num, $limit = null)
     {
+        if ($num < 1) {
+            throw new InvalidArgumentException('Pages should start at 1.');
```

Comment:
should or must?

Reply:
I copy this from docs

Reply:
also maybe we want to change 
`$offset = ($num - 1) * $limit;` to `$offset = ($num ) * $limit;` in future

Reply:
What I am getting at: With your added exception it sounds more logical to make this a "must" here, as should implies it work also work with 0 etc.